Mathlib/RingTheory/DedekindDomain/Ideal.lean

```diff
@@ -1538,4 +1538,40 @@ theorem multiplicity_normalizedFactorsEquivSpanNormalizedFactors_symm_eq_multipl
     multiplicity_normalizedFactorsEquivSpanNormalizedFactors_eq_multiplicity hr ha]
 #align multiplicity_normalized_factors_equiv_span_normalized_factors_symm_eq_multiplicity multiplicity_normalizedFactorsEquivSpanNormalizedFactors_symm_eq_multiplicity
 
+theorem count_normalizedFactors_eq_count_normalizedFactors_span {R : Type _} [CommRing R]
+    [IsDomain R] [IsPrincipalIdealRing R] [NormalizationMonoid R] [DecidableEq R]
+    [DecidableEq (Ideal R)] {r X : R} (hr : r ≠ 0) (hX₀ : X ≠ 0) (hX₁ : normUnit X = 1)
```

Comment:
Could you drop `hX₁` by inserting a `normalize` call in the right place in the conclusion, and perhaps derive this theorem from it?